docs/source/api.rst

```diff
@@ -104,110 +115,80 @@ For special applications, the
 is experimental, see :ref:`dask` for more details. It might use threading as
 well, depending on the Dask scheduler that is used by :code:`compute()`.
 
+.. _`pipelined`:
 
-Common executor choices
-.......................
-
-.. versionadded:: 0.9.0
-
-:meth:`libertem.api.Context.make_with` provides a convenient shortcut to start a
-:class:`~libertem.api.Context` with common executor choices. See the
-:meth:`API documentation <libertem.api.Context.make_with>`
-for available options!
-
-Connect to a cluster
-....................
-
-See :ref:`cluster` on how to start a scheduler and workers.
+Pipelined executor
+------------------
 
-.. Not run with docs-check since it doesn't play well with launching a multiprocessing
-   cluster
+.. versionadded:: 0.10.0
 
-.. code-block:: python
+For live data processing using
+`LiberTEM-live <https://libertem.github.io/LiberTEM-live/>`_, the
+:class:`~libertem.executor.pipelined.PipelinedExecutor`
+provides a multiprocessing executor that routes the live data source in a round-robin
+fashion to worker processes. This is important to support processing that cannot keep
+up with the detector speed on a single CPU core. This executor also works for offline
+data sets in principle, but is not optimized for that use case.
 
-    from libertem import api
-    from libertem.executor.dask import DaskJobExecutor
+Please see :ref:`pipelined executor` for a reference of the pipelined executor,
+and `the LiberTEM-live documentation <https://libertem.github.io/LiberTEM-live/>`_
+for details on live processing.
 
-    # Connect to a Dask.Distributed scheduler at 'tcp://localhost:8786'
-    with DaskJobExecutor.connect('tcp://localhost:8786') as executor:
-        ctx = api.Context(executor=executor)
-        ...
 
 .. _`cluster spec`:
 
-Customize CPUs and CUDA devices
-...............................
+Specifying executor type, CPU and GPU workers
+.............................................
 
-To control how many CPUs and which CUDA devices are used, you can specify them as follows:
+.. versionadded:: 0.9.0
```

Comment:
Might need a `.. versionchanged:: 0.12.0` in addition, noting the new `gpus`/`cpus` parameters